instruqt-tracks/DEVELOPMENT.md

```diff
@@ -0,0 +1,46 @@
+# How to safely test Instruqt tracks
+
+Instruqt can be a bit finicky when it comes to testing tracks, because
+it appears internally to use track and challenge IDs and embeds them
+in your `track.yml` and `assigment.md` files when you push tracks. This
+can cause a problem when trying to make a test version of the track; if
+you are not careful you can confuse Instruqt or cause it to overwrite
+the production version of a track. 
+
+What follows is our standard flow and procedure for making a test 
+version of the track.
+
+# Standard Flow
+
+1. Create a Jira ticket for your work—all work should be logged and
+   documented in a Jira ticket anyways, and we use the ticket identifier
+   to provide context when committing changes. For this document, we're
+   going to assume you have a ticket number of `IL-99999`
+1. Ensure you are on the `master` branch of this repo (the current default
+   branch) and do not have uncommited changes. Pull the latest from 
+   `origin` (e.g. `git pull origin master`)
+1. Create a branch for this work, e.g. `git checkout -b IL-99999`
+1. In the track directory, run the command
+   `make jira=IL-99999 new_test_track`
```

Comment:
I wonder if we can make this simpler for people, i.e. if the branch name looks like a JIRA ticket number, and there's no `jira=...` argument, default to assuming the branch name is the ticket number.

Reply:
Oh, that's a good idea. If it's, currently, `IL-[0-9]+` for the branch name then just use that if `$(jira)` isn't set. I won't block this merge on that but put it in the back of my head for future enhancement.